python-avd/pyavd/_eos_designs/structured_config/base/platform_mixin.py

```diff
@@ -0,0 +1,129 @@
+from typing import TYPE_CHECKING, Protocol
+
+from pyavd._eos_cli_config_gen.schema import EosCliConfigGen
+from pyavd._eos_designs.structured_config.structured_config_generator import structured_config_contributor
+from pyavd._errors import AristaAvdInvalidInputsError
+from pyavd.j2filters import list_compress, range_expand
+
+if TYPE_CHECKING:
+    from pyavd._eos_designs.schema import EosDesigns
+
+    from . import AvdStructuredConfigBaseProtocol
+
+
+class PlatformMixin(Protocol):
+    """
+    Mixin Class used to generate structured config for one key.
+
+    Class should only be used as Mixin to a AvdStructuredConfig class or other Mixins.
+    """
+
+    @structured_config_contributor
+    def platform(self: AvdStructuredConfigBaseProtocol) -> None:
+        """
+        Platform set based on platform data-model, using various platform related information.
+
+        platform set is based on the following:
+
+        * platform_settings.lag_hardware_only,
+        * platform_settings.trident_forwarding_table_partition and switch.evpn_multicast facts
+        * data_plane_cpu_allocation_max,
+        * platform_settings.feature_support.platform_sfe_interface_profile.supported
+        """
+        if (lag_hardware_only := self.shared_utils.platform_settings.lag_hardware_only) is not None:
+            self.structured_config.platform.sand.lag.hardware_only = lag_hardware_only
+
+        trident_forwarding_table_partition = self.shared_utils.platform_settings.trident_forwarding_table_partition
+        if trident_forwarding_table_partition and self.shared_utils.evpn_multicast:
+            self.structured_config.platform.trident.forwarding_table_partition = trident_forwarding_table_partition
+
+        if (cpu_max_allocation := self.shared_utils.node_config.data_plane_cpu_allocation_max) is not None:
+            self.structured_config.platform.sfe.data_plane_cpu_allocation_max = cpu_max_allocation
+        elif self.shared_utils.is_wan_server:
+            # For AutoVPN Route Reflectors and Pathfinders, running on CloudEOS, setting
+            # this value is required for the solution to work.
+            msg = "For AutoVPN RRs and Pathfinders, 'data_plane_cpu_allocation_max' must be set"
+            raise AristaAvdInvalidInputsError(msg)
+
+        # populate interface profile for SFE platform (if supported)
+        if self.shared_utils.is_sfe_interface_profile_supported and (sfe_member_interfaces_for_profile := self._get_sfe_interface_profile_member_interfaces):
+            default_sfe_interface_profile_name = "Default_Interface_Profile"
+            # build single ProfilesItem and append to Profiles
+            profiles_item = EosCliConfigGen.Platform.Sfe.Interface.ProfilesItem(
+                name=default_sfe_interface_profile_name,
+                interfaces=sfe_member_interfaces_for_profile,
+            )
+            self.structured_config.platform.sfe.interface.profiles.append(profiles_item)
```

Comment:
we can just use append_new as you don't modify afterwards
```suggestion
            default_sfe_interface_profile_name = "Default_Interface_Profile"
            # build single ProfilesItem and append to Profiles
            self.structured_config.platform.sfe.interface.profiles.append_new(
                            name=default_sfe_interface_profile_name,
                interfaces=sfe_member_interfaces_for_profile,
            )
```